Tests/AsyncAlgorithmsTests/TestZipLatestFrom.swift

```diff
@@ -0,0 +1,782 @@
+//
+//  TestZipLatestFrom.swift
+//  
+//
+//  Created by Thibault Wittemberg on 01/04/2022.
+//
+
+@preconcurrency import AsyncAlgorithms
+@preconcurrency import XCTest
+
+/// This spy will call the `onLatestKnownElement` function when `next()` is called but it will pass the latest known produced element from `Base` as an argument.
+/// We can use the `onLatestKnownElement` callback to fulfill expectations when we need to be sure an element from `Base` has been consumed.
+fileprivate struct AsyncSpySequence<Base: AsyncSequence>: AsyncSequence {
+  typealias Element = Base.Element
+  typealias AsyncIterator = Iterator
+
+  let base: Base
+  let onLatestKnownElement: (Base.Element) async -> Void
+
+  init(
+    _ other: Base,
+    onLatestKnownElement: @escaping (Base.Element) async -> Void
+  ) {
+    self.base = other
+    self.onLatestKnownElement = onLatestKnownElement
+  }
+
+  func makeAsyncIterator() -> AsyncIterator {
+    Iterator(
+      base: self.base.makeAsyncIterator(),
+      onLatestKnownElement: self.onLatestKnownElement
+    )
+  }
+
+  struct Iterator: AsyncIteratorProtocol {
+    var base: Base.AsyncIterator
+    var lastKnownElement: Element? = nil
+    let onLatestKnownElement: (Base.Element) async -> Void
+
+    mutating func next() async rethrows -> Element? {
+      if let nonNilLastKnownElement = self.lastKnownElement {
+        await self.onLatestKnownElement(nonNilLastKnownElement)
+      }
+
+      let baseElement = try await self.base.next()
+      self.lastKnownElement = baseElement
+      return baseElement
+    }
+  }
+}
+
+final class TestZipLatestFrom: XCTestCase {}
+
+// MARK: test for AsyncZipLatestSequence
+extension TestZipLatestFrom {
```

Comment:
it would be good to get some coverage numbers so that we can get an idea of how much is addressed in the implementation by these tests; generally we aimed to have 90%+ coverage. Obviously coverage does not mean we cover ALL permutations of usage but it definitely assuages stability concerns etc.

Reply:
I'm having some trouble getting the coverage in Xcode because of an error "Failed to merge raw profiles ...". Does that ring a bell?